src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -272,9 +272,27 @@ private function addProfilerSection(ArrayNodeDefinition $rootNode)
                         ->booleanNode('collect')->defaultTrue()->end()
                         ->booleanNode('only_exceptions')->defaultFalse()->end()
                         ->booleanNode('only_master_requests')->defaultFalse()->end()
-                        ->scalarNode('dsn')->defaultValue('file:%kernel.cache_dir%/profiler')->end()
-                        ->scalarNode('username')->defaultValue('')->end()
-                        ->scalarNode('password')->defaultValue('')->end()
+                        ->scalarNode('dsn')
+                            ->defaultValue('file:%kernel.cache_dir%/profiler')
+                            ->beforeNormalization()
+                                ->always()
+                                ->thenInvalid('The profiler.dsn configuration key must start with "file:" because all the storages except the filesystem are deprecated since version 2.8 and will be removed in 3.0.')
```

Comment:
That's too strong. We should still accept other storage engines, but trigger a deprecation notice in such cases (same for the username and password).

You can see an example in the same file for `csrf_protection`.